src/index.js

```diff
@@ -0,0 +1,223 @@
+import crypto from 'isomorphic-webcrypto'
+import argon2 from 'argon2-wasm'
+import nacl from 'tweetnacl'
+import bip32 from 'bip32'
+
+const bufferFrom = arr => Buffer.from(arr)
+
+const bufferConcat = arr => Buffer.concat(arr)
+
+const reverse = arr => arr.reverse()
+
+const defaultTo = (any, d) => isUndefined(any) ? d : any
+
+const get = (o, k, d) => {
+  if (isUndefined(o)) return d
+  const r = o[k]
+  return isUndefined(r) ? d : r
+}
+
+const isUndefined = any => typeof any === 'undefined'
+
+const buf2hex = buffer => {
+  return Array.from(new Uint8Array(buffer))
+    .map(b => b.toString(16).padStart(2, '0'))
+    .join('');
+}
+
+const hash = (...args) => {
+  // map args into buffers and concat into one buffer
+  const buffer = bufferConcat(args.map(a => bufferFrom(a)))
+  // generate a SHA-512 hash from input buffer
+  return crypto.subtle.digest({ name: 'SHA-512' }, buffer)
+}
+
+
+
+const argon2u = (entropy, ticketSize) => argon2({
+  pass: entropy, // string or Uint8Array
+  salt: 'urbitkeygen',
+  type: 10, // argon2.ArgonType.Argon2u,
+  hashLen: ticketSize,
+  // distPath: 'node_modules/argon2-wasm/dist',
+  parallelism: 4,
+  mem: 512000,
+  time: 1,
+})
+
+
+
+const childSeedFromSeed = async config =>  {
+  const { seed, type, revision, ship, password } = config
+
+  const salt = typeof ship === 'number'
+    ? `${salt}-${ship}`
+    : `${type}-${revision}`
```

Comment:
This... doesn't match the previous implementation, I don't think? On line 54, `salt` is still empty, correct? We want `type-revision` always, and then `-ship` if we have one. If I'm reading this correctly it just gives you either `-ship` or `type-revision`.

Reply:
You're right, this isn't correct.